UserTools/Configuration/Configuration.cpp

```diff
@@ -72,15 +74,31 @@ void Configuration::Thread(Thread_args* arg){
 
 bool Configuration::LoadConfig(){
 
+  //m_data->services->SendLog("Loading configuration "+std::to_string(m_data->run_configuration)+" from DB" , 1);
+  Log("Loading configuration "+std::to_string(m_data->run_configuration)+" from DB" , 1);
   std::string config_json="";
+  std::string response="";
   if(m_data->services->GetRunDeviceConfig(config_json, m_data->run_configuration)){
+    
     m_data->vars.JsonParser(config_json);
-    m_data->change_config=true;
-    InitialiseConfiguration("");
+    
+    InitialiseConfiguration(m_configfile);
     ExportConfiguration();
-    return true;
+    std::cout<<m_tool_name<<" m_variables:\n--------------\n";
+    m_variables.Print();
+    std::cout<<"--------------\n";
+    
+    m_data->change_config=true;
+    
+  } else {
+    
+    m_data->services->SendLog("ERROR "+m_tool_name+": Failed to load config from DB with error '"+config_json+"'" , 0);
```

Comment:
i wonder if we should let it try 5 times or something then give up..... could be implimented on the calling side. however throwing alarm on first try might be too much

Reply:
if the retrieval mechanism isn't robust enough, those changes (such as retries) should be implemented there. The caller should not need to call a function more thane once.